app/src/main/java/com/github/kulebin/myfoursquareapp/MainActivity.java

```diff
@@ -1,42 +1,41 @@
 package com.github.kulebin.myfoursquareapp;
 
 import android.os.Bundle;
+import android.support.v4.app.FragmentManager;
 import android.support.v7.app.AppCompatActivity;
-import android.support.v7.widget.LinearLayoutManager;
-import android.support.v7.widget.RecyclerView;
-import android.view.View;
-import android.widget.Toast;
+import android.view.MenuItem;
 
-import com.github.kulebin.myfoursquareapp.view.IViewCallback;
-import com.github.kulebin.myfoursquareapp.presenter.VenueListPresenter;
-
-public class MainActivity extends AppCompatActivity implements IViewCallback {
-
-    private final VenueListPresenter mListPresenter = new VenueListPresenter(this);
+public class MainActivity extends AppCompatActivity implements VenueListFragment.Callback {
 
     @Override
     protected void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        initRecyclerView();
-        mListPresenter.onViewCreated();
-    }
-
-    private void initRecyclerView() {
-        final RecyclerView recyclerView = (RecyclerView) findViewById(R.id.venueRecyclerView);
-        recyclerView.setLayoutManager(new LinearLayoutManager(this));
-        recyclerView.setAdapter(mListPresenter.getVenueListAdapter());
+        final FragmentManager fm = getSupportFragmentManager();
```

Comment:
We can get such case after the configuration was changed (e.g. after screen rotaion)